recipes/cpprestsdk/all/conanfile.py

```diff
@@ -40,6 +46,10 @@ def _build_subfolder(self):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        else:
```

Comment:
Since the default values above only apply on Windows, would it be better to remove `"pplx_impl": "win"`, etc. from `default_options` and in `config_options` replace the `else` with the following code under the `if self.settings.os == "Windows":` branch?
```
            if self.options.get_safe("pplx_impl") == None:
                self.options["pplx_impl"] = "win"
            if self.options.get_safe("http_client_impl") == None:
                self.options["http_client_impl"] = "winhttp"
            if self.options.get_safe("http_listener_impl") == None:
                self.options["http_listener_impl"] = "httpsys"

Reply:
You can't remove default options, Conan will break when building. 

It can be improved:
```
if self.options.get_safe("http_listener_impl") == None:
    self.options["http_listener_impl"] = "httpsys"
```

```
if not self.options.get_safe("http_listener_impl"):
    self.options["http_listener_impl"] = "httpsys"
```

Reply:
@uilianries

> You can't remove default options, Conan will break when building.

I read this: https://docs.conan.io/en/latest/reference/conanfile/attributes.html#default-options which says "You can also set the options conditionally to a final value with `config_options()` instead of using `default_options`". I was asking if that would be more appropriate here?

Reply:
You can remove self.options, not self.default_options.